validator/client/validator.go

```diff
@@ -418,3 +396,60 @@ func (v *validator) domainData(ctx context.Context, epoch uint64, domain []byte)
 
 	return res, nil
 }
+
+func (v *validator) logDuties(slot uint64, duties []*ethpb.DutiesResponse_Duty) {
+	attesterKeys := make([][]string, params.BeaconConfig().SlotsPerEpoch)
+	for i := range attesterKeys {
+		attesterKeys[i] = make([]string, 0)
+	}
+	proposerKeys := make([]string, params.BeaconConfig().SlotsPerEpoch)
+	slotOffset := (slot / params.BeaconConfig().SlotsPerEpoch) * params.BeaconConfig().SlotsPerEpoch
+
+	for _, duty := range duties {
+		if v.emitAccountMetrics {
+			fmtKey := fmt.Sprintf("%#x", duty.PublicKey)
+			validatorStatusesGaugeVec.WithLabelValues(fmtKey).Set(float64(duty.Status))
+		}
+
+		// Only interested in validators who are attesting/proposing.
+		// Note that SLASHING validators will have duties but their results are ignored so we don't bother with them.
+		if duty.Status != ethpb.ValidatorStatus_ACTIVE && duty.Status != ethpb.ValidatorStatus_EXITING {
+			continue
+		}
+
+		validatorKey := fmt.Sprintf("%#x", bytesutil.Trunc(duty.PublicKey))
+		attesterIndex := duty.AttesterSlot - slotOffset
+		if attesterIndex >= params.BeaconConfig().SlotsPerEpoch {
+			log.WithField("duty", duty).Warn("Invalid attester slot")
+		} else {
+			attesterKeys[duty.AttesterSlot-slotOffset] = append(attesterKeys[duty.AttesterSlot-slotOffset], validatorKey)
+		}
+
+		if duty.ProposerSlot != 0 {
+			proposerIndex := duty.ProposerSlot - slotOffset
+			if proposerIndex >= params.BeaconConfig().SlotsPerEpoch {
+				log.WithField("duty", duty).Warn("Invalid proposer slot")
+			} else {
+				proposerKeys[proposerIndex] = validatorKey
+			}
+		}
+		// 0.11 has multiple proposer slots; use this code in place of the above when merged.
```

Comment:
Please fix

Reply:
Updated; will run locally for a bit to ensure the manual bits of the merge are correct.